modules/graphics/src/mrb_texture.cxx

```diff
@@ -0,0 +1,82 @@
+#include <mruby.h>
+#include <mruby/class.h>
+#include <mruby/data.h>
+#include <mruby/numeric.h>
+#include <mruby/string.h>
+#include "moon/mrb/texture.hxx"
+#include "moon/texture.hxx"
+#include "moon/api.h"
+
+static struct RClass *texture_class;
+
+static void
+texture_free(mrb_state *mrb, void *p)
+{
+  Moon::Texture *texture = (Moon::Texture*)p;
+  if (texture) {
+    delete(texture);
+  }
+}
+
+MOON_C_API const struct mrb_data_type texture_data_type = { "Texture", texture_free };
+
+static inline Moon::Texture*
+get_texture(mrb_state *mrb, mrb_value self)
+{
+  return (Moon::Texture*)mrb_data_get_ptr(mrb, self, &texture_data_type);
+}
+
+MOON_C_API mrb_value
+mmrb_texture_load_file(mrb_state *mrb, const char *filename)
+{
+  mrb_value fn = mrb_str_new_cstr(mrb, filename);
+  return mrb_obj_new(mrb, texture_class, 1, &fn);
+}
+
+static mrb_value
+texture_initialize(mrb_state *mrb, mrb_value self)
+{
+  char* filename;
+  Moon::Texture *texture;
+  mrb_get_args(mrb, "z", &filename);
+
+  texture = (Moon::Texture*)DATA_PTR(self);
+  if (texture) {
+    texture_free(mrb, (void*)texture);
+    texture = NULL;
+  }
+  DATA_PTR(self) = NULL;
+
+  if (exists(filename)) {
+    texture = new Moon::Texture(filename);
+  } else {
+    mrb_raisef(mrb, E_SCRIPT_ERROR,
+               "cannot load such file -- %S",
+               mrb_str_new_cstr(mrb, filename));
+  }
+  mrb_data_init(self, texture, &texture_data_type);
+  return self;
+}
+
+static mrb_value
+texture_width(mrb_state *mrb, mrb_value self)
+{
+  return mrb_float_value(mrb, get_texture(mrb, self)->GetWidth());
+}
+
+static mrb_value
+texture_height(mrb_state *mrb, mrb_value self)
+{
+  return mrb_float_value(mrb, get_texture(mrb, self)->GetHeight());
+}
+
+MOON_C_API void
+mmrb_texture_init(mrb_state *mrb, struct RClass* mod)
+{
+  texture_class = mrb_define_class_under(mrb, mod, "Texture", mrb->object_class);
+  MRB_SET_INSTANCE_TT(texture_class, MRB_TT_DATA);
+
+  mrb_define_method(mrb, texture_class, "initialize", texture_initialize, MRB_ARGS_REQ(1));
+  mrb_define_method(mrb, texture_class, "w",      texture_width,      MRB_ARGS_NONE());
```

Comment:
I removed all the aliases, every last one of them, it is forever `w`/`h` from now on.

Reason: While hacking away one day, I forgot that w / h was aliased as width / height, overwrote the width / height methods to add callbacks and went along using the aliased w / h instead of the callback tied width / height. 
In the end, I removed all width / height methods, all aliases and just stuck with everything having a w / h, since its much easier to type and i'm less likely to typo heigth... height*